include/NeoFOAM/fields/BoundaryFields.hpp

```diff
@@ -0,0 +1,128 @@
+class BoundaryFields
+{
+
+public:
+
+    /**
+     * @brief Copy constructor.
+     * @param rhs The boundaryFields object to be copied.
+     */
+    BoundaryFields(const BoundaryFields<T>& rhs)
+        : exec_(rhs.exec_), value_(rhs.value_), refValue_(rhs.refValue_),
+          valueFraction_(rhs.valueFraction_), refGrad_(rhs.refGrad_),
+          boundaryTypes_(rhs.boundaryTypes_), offset_(rhs.offset_), nBoundaries_(rhs.nBoundaries_),
+          nBoundaryFaces_(rhs.nBoundaryFaces_)
+    {}
+
+
+    BoundaryFields(const Executor& exec, int nBoundaryFaces, int nBoundaries)
+        : exec_(exec), value_(exec, nBoundaryFaces), refValue_(exec, nBoundaryFaces),
+          valueFraction_(exec, nBoundaryFaces), refGrad_(exec, nBoundaryFaces),
+          boundaryTypes_(exec, nBoundaries), offset_(exec, nBoundaries + 1),
+          nBoundaries_(nBoundaries), nBoundaryFaces_(nBoundaryFaces)
+    {}
+
+    /**
+     * @brief Get the view storing the computed values from the boundary
+     * condition.
+     * @return The view storing the computed values.
+     */
+    const NeoFOAM::Field<T>& value() const { return value_; }
+
+    NeoFOAM::Field<T>& value() { return value_; }
+
+    /**
+     * @brief Get the view storing the Dirichlet boundary values.
+     * @return The view storing the Dirichlet boundary values.
+     */
+    const NeoFOAM::Field<T>& refValue() const { return refValue_; }
+
+    NeoFOAM::Field<T>& refValue() { return refValue_; }
+
+    /**
+     * @brief Get the view storing the fraction of the boundary value.
+     * @return The view storing the fraction of the boundary value.
+     */
+    const NeoFOAM::Field<scalar>& valueFraction() const { return valueFraction_; }
+
+    NeoFOAM::Field<scalar>& valueFraction() { return refValue_; }
+
+    /**
+     * @brief Get the view storing the Neumann boundary values.
+     * @return The view storing the Neumann boundary values.
+     */
+    const NeoFOAM::Field<T>& refGrad() const { return refGrad_; }
+
+    NeoFOAM::Field<T>& refGrad() { return refGrad_; }
+
+    /**
+     * @brief Get the view storing the boundary types.
+     * @return The view storing the boundary types.
+     */
+    const NeoFOAM::Field<int>& boundaryTypes() const { return boundaryTypes_; }
+
+    /**
+     * @brief Get the view storing the offsets of each boundary.
+     * @return The view storing the offsets of each boundary.
+     */
+    const NeoFOAM::Field<localIdx>& offset() const { return offset_; }
+
+    /**
+     * @brief Get the number of boundaries.
+     * @return The number of boundaries.
+     */
+    int nBoundaries() { return nBoundaries_; }
+
+    /**
+     * @brief Get the number of boundary faces.
+     * @return The number of boundary faces.
+     */
+    int nBoundaryFaces() { return nBoundaryFaces_; }
+
+    const Executor& exec() { return exec_; }
+
+private:
+
+    Executor exec_;
+    NeoFOAM::Field<T> value_;              ///< The view storing the computed values from the
```

Comment:
I think view is misleading here.
```suggestion
    NeoFOAM::Field<T> value_;              ///< The Field storing the computed values from the
```
Also in the other ocurrances.